packages/commonwealth/client/scripts/stores/zustand/sidebar.ts

```diff
@@ -0,0 +1,27 @@
+import { create, StateCreator } from 'zustand';
+import { SidebarMenuName } from 'views/components/sidebar';
+import { createStore } from 'zustand/vanilla';
+
+export interface SidebarSlice {
+  menuName: SidebarMenuName;
+  toggled: boolean;
+  toggle: (val: boolean) => void;
+  setMenu: (menuName: SidebarMenuName) => void;
+}
+
+const createSidebarSlice: StateCreator<SidebarSlice, [], [], SidebarSlice> = (
+  set,
+) => ({
+  menuName: 'default',
+  toggled: false,
+  toggle: (val) => set(() => ({ toggled: val })),
+  setMenu: (menuName) => set(() => ({ menuName })),
+});
+
+export const sidebarStore = createStore<SidebarSlice>((...args) => ({
+  ...createSidebarSlice(...args),
```

Comment:
option 2: small store that is only about sidebar

Reply:
I much prefer small stores per state regions. IMO we should not have a global store or we reproduce the same "junk drawer" pattern as we currently have. This also allows us to more cleanly separate imports relating to different parts of the codebase.